
# Plain Vanila Bonds




- Plain vanilla bonds are conventional bonds that pay a fixed rate of interest.

Cash Flows for a Plain Vanilla Bond:

In a traditional (option-free) fixed-rate bond, the fixed payments are the promised future cash flows are a series of coupon interest payments and repayment of the full principal at maturity. Coupon payments are scheduled for example annually - annual payments for a fixed number of years with the final coupon to be paid together with the full principal on the maturity date. 

For a plain vanilla bond, periodic coupon payments do not change during the bond’s life and the entire payment of principal occurs at maturity. This payment structure has been adopted for majority of government and corporate bonds, and it is known as bullet bonds.



### Price Valuation:

Bond's price at issuance is the present value (PV) of the future cash flows discounted at the market discount rate - the rate of return required by investors given the risk of the investment in the bond aka required yield or required rate of return). 

Below is the formula for calculating bond's price using the time-value-of-money calculation to obtain the present value (PV) for a market discount rate:



\begin{aligned}Price &={\begin{matrix}\left({\frac {C}{1+r}}+{\frac {C}{(1+r)^{2}}}+...+{\frac {C}{(1+r)^{n}}}\right)+{\frac {M}{(1+r)^{n}}}\end{matrix}}\\&={\begin{matrix}\left(\sum _{i=1}^{n}{\frac {C}{(1+r)^{i}}}\right)+{\frac {M}{(1+r)^{n}}}\end{matrix}}\\&={\begin{matrix}C\left({\frac {1-(1+r)^{- n}}{r}}\right)+M(1+r)^{-n}\end{matrix}}\end{aligned}


where:

Price = market price of bond

M = par or face value or the principal amount of the bond (the amount the issuer agrees to repay the bondholders at maturity)

n= number of periodic payments

c = coupon rate or the nominal rate of a bond

C = c * M = bond’s coupon or the fixed amount of periodic coupon payments

r = market discount rate or the required rate return (i.e.,  required yield or observed/appropriate yield to maturity)


$ = currency denomination (however, bonds can be issued in any currency, and there are dual-currency bonds and currency option bonds that give issuers/bondholders the right to choose the currency in which they want to pay/receive interest payments and the principal) 


## Time Money Valuation



The concept of Time Value of Money is based on the idea that:

*A dollar in hand today is worth more than a dollar to be received tomorrow.*


Time value of money is of important consideration in making investment decisions.  In the bond's price formula above, the first term - the discount interest calculations is used to compute the present value (PV) of the timing of the promised future cash flows of the coupon payments. Explanation:

$C$ = fixed-coupon payment. However, interest received means that interest earns interest, so:

$C_1$ = the amount of money at the end of year 1: 

= fixed-coupon payment plus interest: $ = C + rC = C(1 + r)$


$C_2$ = the amount of money at the end of year 2: 

$= C_{1}(1 + r) = C(1+ r)(1 + r) = C(1+ r)^2$

...

$C_n$ = the amount of money at the end of year n: 

$= C_{n-1}(1 + r) = C(1+ r)^n$


### Defining  Bullet Bond Class:

In [1]:
%%writefile b1.py

import math 

class BulletBond:
    def __init__(self, n, M, c):
        self.n = n
        self.M = M
        self.c = c

    def Price(self, r): 
        return self.c * self.M *(1 - math.pow(1+r,-self.n))/r + self.M * math.pow(1+r,-self.n)
    
    

Overwriting b1.py


In [2]:
import b1

### Example :

Suppose the coupon rate on a bond is 5% and the payment is made once a year. If the time-to-maturity is 5 years and the market discount rate is 6%. Then, the price of the bond is $95.787 per 100 of par value as shown below:

In [3]:
price = b1.BulletBond(5,100,0.04).Price(0.06)
price

91.57527242886857

In practice, bond prices are quoted as a percentage of their par value. In this example, the quote price is the same 91.575.

Because the bond's price is below par value, this bond is said to be **trading at a discount**

Upon maturity, the investor gains:

In [4]:
(4*5 + 100)- price 

28.424727571131427

Rounded the gain on investment is: $28.42

### Example 2:  

Suppose that another five-year bond has a coupon rate of 6% paid annually. If the market discount rate is 4%, the price of the bond is $108.903 as shown:

In [5]:
price = b1.BulletBond(5,100,0.06).Price(0.04)
price

108.9036446620324

This bond is **trading at a premium** because its price is above par value!

Example 3:
Suppose that another five-year bond has a coupon rate of 4% paid annually and the market discount rate equal to the coupon rate. The price of the bond is $100 as shown:

In [6]:
price = b1.BulletBond(5,100,0.04).Price(0.04)
price

100.0

This bond is **trading at par** because the bond is priced at par value.

### Summary:  


#### Coupon rate  vs Market discount rate

c -  the coupon rate indicates the amount the issuer promises to pay the bondholders each year in interest

r -  the market discount rate reflects the amount investors need to receive in interest each year in order to pay full par value for the bond

The two examples demonstrate that the price of a fixed-rate bond relative to par value, depends on the relationship of the coupon rate to the market discount rate. That is:


- When c < r : the bond is priced at a *discount* below par value

- When c > r : the bond is priced at a *premium* above par value.

- When c = r , the bond is priced *at par* value.